### DISCUSSION TITLE
Updating a Heroku deployment

### DISCUSSION THREAD

**Discussion by douglasac10** [pglombardo/PasswordPusher#964] [Q&A] [answered]
Hi,

We have an instance of Password Pusher that we have running on a Heroku deployment. We absolutely love it!

What is the best way of updating this one?

Thanks!

**Comment by pglombardo** [comment, marked as answer]
Hi @douglasac10 - thanks for the nice words!

It depends on how you have it setup but most common way to update your Heroku deployment from the CLI (using the `heroku` command) is:

```bash
git checkout master
git pull origin --rebase
git push heroku master

# Then run the database migration to update the database tables (no data is lost)
heroku run rails db:migrate
```

Before you do that, it's always good to know/have some safeties:

1. Backup and export the database first: `heroku pg:backups:capture --app application-name`
2. Know about releases and how to rollback in case of error: `heroku releases` && `heroku rollback`

Also check the latest version of the [configuration document](https://github.com/pglombardo/PasswordPusher/blob/master/Configuration.md).  There are some new options that you may want to set.  You can add these environment variables with `heroku config:set PWP__DEFAULT_LOCALE =es`.  See also the [heroku docs](https://devcenter.heroku.com/articles/config-vars).

I hope this helps!

**Reply by douglasac10** [reply]
Thanks for that - worked a treat!

I did have to add the variable BUNDLE_WITH=postgres to the Config Vars ([as mentioned here](https://github.com/pglombardo/PasswordPusher/issues/441#issuecomment-1409401413)), but once that was done all good :)

**Reply by pglombardo** [reply]
Doh good point - that step is new in the last few months.  Glad to hear everything is updated!
